x-pack/apm-server/sampling/processor.go

```diff
@@ -446,11 +451,11 @@ func (p *Processor) Run() error {
 				remoteDecision = true
 			case traceID = <-localSampledTraceIDs:
 			}
-			if err := p.storage.WriteTraceSampled(traceID, true); err != nil {
+			if err := ignoreENOSPCErr(p.storage.WriteTraceSampled(traceID, true)); err != nil {
```

Comment:
Perhaps instead of ignoring some errors, we should log a warning and continue? At least then we can index what we have.